src/backend/distributed/sql/udfs/citus_locks/latest.sql

```diff
@@ -0,0 +1,84 @@
+-- citus_locks combines the pg_locks views from all nodes and adds global_pid, and nodeid.
+-- The columns of citus_locks don't change based on the Postgres version, however the
+-- pg_locks's columns do. Postgres 14 added one more column to pg_locks (waitstart
+-- timestamptz). citus_locks has the most expansive column set, including the newly added
+-- column. If citus_locks is queried in a Postgres version where pg_locks doesn't have
+-- some columns, the values for those columns in citus_locks will be NULL
+CREATE OR REPLACE FUNCTION pg_catalog.citus_locks (
```

Comment:
does this show locks for SELECT query in the following case:

```
BEGIN;
-- on the coordinator
ALTER TABLE ADD COLUMN X INT;
```

```SQL
-- on the worker
psql -p worker_port -c "SELECT * FROM test"
```

any node: `SELECT * FROM citus_locks`